docker/release/build-image-multi-arch.sh

```diff
@@ -0,0 +1,155 @@
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+    esac
+done
+
+# Validate the required parameters to present
+if [ -z "$VERSION" ] || [ -z "$DOCKERFILE" ] || [ -z "$PRODUCT" ] || [ -z "$ARCHITECTURE" ] || [ -z "$REPOSITORY" ]; then
+  echo "You must specify '-v VERSION', '-f DOCKERFILE', '-p PRODUCT', '-a ARCHITECTURE', '-r REPOSITORY'"
+  usage
+  exit 1
+else
+  echo $VERSION $DOCKERFILE $PRODUCT $ARCHITECTURE
+  IFS=', ' read -r -a ARCHITECTURE_ARRAY <<< "$ARCHITECTURE"
+  IFS=', ' read -r -a TARBALL_ARRAY <<< "$TARBALL"
+fi
+
+if [ "$PRODUCT" != "opensearch" ] && [ "$PRODUCT" != "opensearch-dashboards" ]
+then
+    echo "Enter either 'opensearch' or 'opensearch-dashboards' as product name for -p parameter"
+    exit 1
+fi
+
+for ARCH in $ARCHITECTURE_ARRAY
+do
+  if [ "$ARCH" != "x64" ] && [ "$ARCH" != "arm64" ]
+  then
+	echo "We only support 'x64' and 'arm64' as architecture name for -a parameter"
+    	exit 1
+  fi
+done
+
+# Warning docker desktop
+if (! docker buildx version)
+then
+    echo -e "\n* You MUST have Docker Desktop to use buildx for multi-arch images."
+    exit 1
+fi
+
+# Prepare docker buildx
+trap cleanup_all TERM INT EXIT
+DIR=`Temp_Folder_Create`
+echo New workspace $DIR
+echo -e "\n* Prepare docker buildx"
+docker buildx use default
+docker buildx create --name $BUILDER_NAME --use
+docker buildx inspect --bootstrap
+
+# Check buildx status
+echo -e "\n* Check buildx status"
+docker buildx ls | grep $BUILDER_NAME
+docker ps | grep $BUILDER_NAME
+
+
+# Copy configs
+cp -v config/${PRODUCT}/* $DIR/
+cp -v ../../scripts/opensearch-onetime-setup.sh $DIR/
+
+# Copy TGZ
+if [ -z "$TARBALL" ]; then
+    for index in ${!ARCHITECTURE_ARRAY[@]}
+    do
+        arch_uname=`echo ${ARCHITECTURE_ARRAY[$index]} | sed 's/x64/x86_64/g;s/arm64/aarch64/g'`
+        URL="https://artifacts.opensearch.org/releases/bundle/${PRODUCT}/${VERSION}/${PRODUCT}-${VERSION}-linux-${ARCHITECTURE_ARRAY[$index]}.tar.gz"
+        echo -e "\nDownloading ${PRODUCT} arch ${ARCHITECTURE_ARRAY[$index]} version ${VERSION} from ${URL}"
+        curl -f $URL -o ${DIR}/${PRODUCT}-${arch_uname}.tgz || exit 1
+    done
+    ls -l $DIR
+else
+    for index in ${!ARCHITECTURE_ARRAY[@]}
+    do
+        arch_uname=`echo ${ARCHITECTURE_ARRAY[$index]} | sed 's/x64/x86_64/g;s/arm64/aarch64/g'`
+        echo -e "\nCopying ${PRODUCT} arch ${ARCHITECTURE_ARRAY[$index]} version ${VERSION} from ${TARBALL_ARRAY[$index]}"
+        cp -v ${TARBALL_ARRAY[$index]} ${DIR}/${PRODUCT}-${arch_uname}.tgz
+    done
+    ls -l $DIR
+fi
+
+# Build multi-arch images
+PLATFORMS=`echo "${ARCHITECTURE_ARRAY[@]/#/linux/}" | sed 's/x64/amd64/g;s/ /,/g'` && echo PLATFORMS $PLATFORMS
+docker buildx build --platform $PLATFORMS --build-arg VERSION=$VERSION --build-arg BUILD_DATE=`date -u +%Y-%m-%dT%H:%M:%SZ` -t ${REPOSITORY}:${VERSION} -t ${REPOSITORY}:latest -f $DOCKERFILE --push $DIR
+
```

Comment:
We should consider writing these in Python, it's a lot less error prone and would be 5-10 lines of code.

The first 100 lines become 5 lines of `argparse`.
The next 100 lines become 3 classes.
It would also be testable.